packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -4441,6 +4429,31 @@ export class ThinEngine {
         );
     }
 
+    /**
+     * Calls to the GL texImage2D and texImage3D functions require three arguments describing the pixel format of the texture.
+     * createTexture derives these from the babylonFormat and useSRGBBuffer arguments and also the file extension of the URL it's working with.
+     * This function encapsulates that derivation for easy unit testing.
+     * @param babylonFormat Babylon's format enum, as specified in ITextureCreationOptions.
+     * @param fileExtension The file extension including the dot, e.g. .jpg.
+     * @param useSRGBBuffer Use SRGB not linear.
+     * @returns The options to pass to texImage2D or texImage3D calls.
+     * @internal
+     */
+    public _getTexImageParametersForCreateTexture(babylonFormat: Nullable<number>, fileExtension: string, useSRGBBuffer: boolean): TexImageParameters {
+        if (babylonFormat === undefined || babylonFormat === null) {
+            babylonFormat = fileExtension === ".jpg" && !useSRGBBuffer ? Constants.TEXTUREFORMAT_RGB : Constants.TEXTUREFORMAT_RGBA;

```

Comment:
This code does not do the same thing than:
```typescript
extension === ".jpg" && !texture._useSRGBBuffer
? gl.RGB
: texture._useSRGBBuffer
? this._glSRGBExtensionValues.SRGB8_ALPHA8
: gl.RGBA;
```
so it would be a breaking change.